resources/Microsoft.Windows.Setting.Language/Microsoft.Windows.Setting.Language.psm1

```diff
@@ -0,0 +1,213 @@
+
+        if (-not $keyExist) {
+            return [Language]::new($this.LocaleName, $false)
+        }
+        
+        return $currentState
+    }
+
+    [void] Set() {
+        if ($this.Test()) {
+            return
+        }
+
+        if ($this.Exist) {
+            # use the LanguagePackManagement module to install the language (requires elevation). International does not have a cmdlet to install language
+            Install-Language -Language $this.LocaleName
+        } else {
+            Uninstall-Language -Language $this.LocaleName
+        }
+    }
+
+    [bool] Test() {
+        $currentState = $this.Get()
+
+        if ($currentState.Exist -ne $this.Exist) {
+            return $false
+        }
+        
+        return $true
+    }
+
+    static [Language[]] Export() {
+        return Get-LocaleList
+    }
+
+    #region Language helper functions
+    static [void] GetInstalledLocality() {   
+        [Language]::InstalledLocality = @{}
+
+        foreach ($locality in [Language]::Export()) {
+            [Language]::InstalledLocality[$locality.LocaleName] = $locality
+        }
+    }
+    #endRegion Language helper functions
+}
+
+<#
+.SYNOPSIS
+    The `DisplayLanguage` DSC Resource allows you to set the display language on your local Windows machine.
+
+.PARAMETER LocaleName
+    The name of the display language. This is the language tag that represents the language. For example, `en-US` represents English (United States).
+
+.PARAMETER Exist
+    Indicates whether the display language should be set. Defaults to $true.
+
+.EXAMPLE
+    PS C:\> Invoke-DscResource -ModuleName Microsoft.Windows.Setting.Language -Name DisplayLanguage -Method Set -Property @{ LocaleName = 'en-US' }
+
+    This example sets the display language to English (United States) on the user.
+#>
+[DscResource()]
+class DisplayLanguage {
+    
+    [DscProperty(Key)]
+    [string] $LocaleName
+
+    [DscProperty()]
+    [bool] $Exist = $true
+
+    hidden [string] $KeyName = 'LocaleName'
+
+    DisplayLanguage() {
+    }
+
+    [DisplayLanguage] Get() {
+        $currentState = [DisplayLanguage]::new()
+
+        # check if user profile contains display language
+        $userProfileLanguageDict = TryGetRegistryValue -Key (Join-Path $global:LocaleUserProfilePath $this.LocaleName) -Property 'CachedLanguageName'
+        if ((TryGetRegistryValue -Key $global:LocaleNameRegistryPath -Property $this.KeyName) -ne $this.LocaleName -and ($null -ne $userProfileLanguageDict)) {
+            $currentState.Exist = $false
+            return $currentState
+        }
+
+        return @{
+            LocaleName = $this.LocaleName
+            Exist      = $true
+        }
+    }
+
+    [void] Set() {
+        if ($this.Test()) {
+            return
+        }
+
+        # TODO: How do we handle sign out and sign in?
+        Set-WinUserLanguageList -Language $this.LocaleName
+
+        # TODO: Exist does not make sense here, we always want a language to exist
```

Comment:
If Exist doesn't make sense here, then you don't need to use it as a parameter. Just use LocaleName as the only parameter.